### ISSUE TITLE
`NoEnvOutsideConfig` fires inside analysed project's own `config/` directory

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#756]
## Description

The [`NoEnvOutsideConfig`](https://psalm.github.io/psalm-plugin-laravel/issues/NoEnvOutsideConfig/) rule fires on `env(...)` calls inside the analysed project's own `config/*.php` files. These are the canonical, documented place to call `env()`, so every Laravel package or application with a `config/` directory gets a wave of false positives.

## Root cause

[`NoEnvOutsideConfigHandler::isInsideConfigDirectory()`](https://github.com/psalm/psalm-plugin-laravel/blob/master/src/Handlers/Rules/NoEnvOutsideConfigHandler.php#L64-L74) compares the analysed file path against `self::$configPath`, which is initialised in [`src/Plugin.php`](https://github.com/psalm/psalm-plugin-laravel/blob/master/src/Plugin.php#L45) from:

```php
NoEnvOutsideConfigHandler::init(
    ApplicationProvider::getApp()->configPath(),
);
```

`ApplicationProvider::getApp()` is the plugin's internal Orchestra Testbench Laravel app. Its `configPath()` returns something under `vendor/orchestra/testbench-core/laravel/config` (relative to wherever the plugin is loaded from), never the config path of the project being analysed. The `str_starts_with` check therefore never matches, and the rule always considers the file "outside the config directory".

## Reproduction

Audited [spatie/laravel-backup](https://github.com/spatie/laravel-backup) with the plugin (v4.8.2, Psalm 7 beta, `errorLevel=1`):

```
ERROR: NoEnvOutsideConfig - config/backup.php:21 - env() called outside config directory.
  'name' => env('APP_NAME', 'laravel-backup'),

ERROR: NoEnvOutsideConfig - config/backup.php:93 - env() called outside config directory.
  env('DB_CONNECTION', 'mysql'),

ERROR: NoEnvOutsideConfig - config/backup.php:185 - env() called outside config directory.
  'password' => env('BACKUP_ARCHIVE_PASSWORD'),
```

Six occurrences in one `config/backup.php` alone.

## Proposed fix

Replace the Testbench-derived comparison with a structural check that matches any path containing a `/config/` segment. The simplest, minimal-risk shape:

```php
private static function isInsideConfigDirectory(string $filePath): bool
{
    return str_contains(
        $filePath,
        DIRECTORY_SEPARATOR . 'config' . DIRECTORY_SEPARATOR,
    );
}
```

This matches:

- Laravel applications (`/my-app/config/app.php`).
- Laravel packages (`/spatie/laravel-backup/config/backup.php`).
- Monorepo packages (`/filamentphp/filament/packages/forms/config/...` if any).

The `init()` call, the `$configPath` field, and the `Plugin.php` wiring can all be removed, since the Testbench-derived path never matched anything useful anyway.

## Impact

Affects every Laravel package or application analysed by the plugin that ships its own `config/*.php` (which is almost all of them). In a single-run audit of `spatie/laravel-backup`, 6 of 333 reported errors trace to this; on larger packages with richer config files the count scales linearly.
